src/frontend/src/app/assets/assets.component.ts

```diff
@@ -1,51 +1,54 @@
 import { Component, OnInit } from '@angular/core';
-import { AssetsService } from '../services/index';
+import { AssetsService, AlertService } from '../services/index';
```

Comment:
can not we use this ALertService as app level import?

Reply:
I did not read any app level concept in Angular4. This is same like java class import. This is all I have learn till now.

Reply:
I meant, when we have it here, then why do we need in each component separately https://github.com/praveen-garg/FogLAMP/blob/frontend/src/frontend/src/app/app.module.ts#L11

It may be okay from angular way that it just only import once even if asked twice

Reply:
We need to see, if its imported at app level then we need here? and does importing at app level should be only for those which are wide-spread and commonly used?

Reply:
There are 3 things in https://github.com/praveen-garg/FogLAMP/blob/frontend/src/frontend/src/app/app.module.ts#L11, `Imports`, ''declarations" and ''providers" .  
1)  **imports**  -> Code/Base modules are imported as building blocks of applications here.

2) **declarations** -> Every custom component need to be declared here. But still need to import the component at the top of the other components we wish to use in.

3) **providers** ->  Use to declare the services as a dependency injection. we can define providers to either the component or the parent module or the app module. A service provided in the app module will have the same instance provided everywhere.  We no longer need to include Service in a **providers** array within the component's metadata. **But We do still need to import the Service at the top of the components that we wish to use**.